pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV2.java

```diff
@@ -27,19 +27,18 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Map.Entry;
-import org.apache.commons.lang3.StringUtils;
 import org.apache.pinot.common.response.ProcessingException;
 import org.apache.pinot.common.utils.DataSchema;
 import org.apache.pinot.common.utils.DataTable;
 import org.apache.pinot.common.utils.StringUtil;
-import org.apache.pinot.core.common.ObjectSerDeUtils;
-import org.apache.pinot.spi.utils.ByteArray;
-import org.apache.pinot.spi.utils.BytesUtils;
 
+import static org.apache.pinot.core.common.datatable.DataTableBuilder.VERSION_2;
+import static org.apache.pinot.core.common.datatable.DataTableUtils.decodeString;
+import static org.apache.pinot.core.common.datatable.DataTableUtils.deserializeDictionaryMap;
+import static org.apache.pinot.core.common.datatable.DataTableUtils.serializeDictionaryMap;
 
-public class DataTableImplV2 implements DataTable {
-  private static final int VERSION = 2;
 
+public class DataTableImplV2 extends DataTableImplBase implements DataTable {
```

Comment:
I think this works but doesn't look correct from OOP point of view. Ideally the abstract base class should implement the interface to make a proper logical hierarchy
DataTable -> DataTableImpleBase (or AbstractDataTableImpl) -> DataTableImplV2, DataTableImplV3

Reply:
Done. Has declare DataTableImpleBase as abstract now and make DataTableImpleBase implement the DataTable interface.